Sources/Classes/RSDeviceModeManager.m

```diff
@@ -141,13 +143,28 @@ - (void) initiateCustomFactories {
             id<RSIntegration> nativeOp = [factory initiate:@{} client:client rudderConfig:self->config];
             [self->integrationOperationMap setValue:nativeOp forKey:factory.key];
             [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSDeviceModeManager: initiateCustomFactories: Initiated custom SDK factory %@", factory.key]];
+            [self handleCallbacks:factory.key withIntegration:nativeOp];
         }
         @catch(NSException* e){
             [RSLogger logError:[[NSString alloc] initWithFormat:@"RSDeviceModeManager: initiateCustomFactories: Exception while initiating custom Factory %@ due to %@", factory.key,e.reason]];
         }
     }
 }
 
+- (void) handleCallbacks:(NSString*)key withIntegration:(id<RSIntegration>)nativeOp {
+    if ([self->integrationCallbacks objectForKey:key] &&
```

Comment:
Should we log an error message and send back nil to the callback when the RSIntegration doesn't respond to the `getUnderlyingInstance` ?

so that this way the person who had registered the callback would be updated about what's happening.

Reply:
Doing so resulting into an error, please refer to [this](https://sonarcloud.io/project/issues?resolved=false&severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&sinceLeakPeriod=true&types=BUG&pullRequest=421&id=rudderlabs_rudder-sdk-ios&open=AYxnyuWONoNNwRIOBT_C) sonarcloud.

Should I remove passing nil to callback? @desusai7

Reply:
Update: This has been solved after removing the faulty `callback(nil)` line.